highfidelity/hifi
# Tweak Teleport + Hand Lasers

Vive-centric PR.  

This PR makes the following changes to teleporting:
- Can no longer teleport while far grabbing objects
- If near grabbing, can only teleport with other hand
- Cannot teleport if pressing movement controls (LX/LY)
- Enter teleport mode instantly
- Enter teleport mode with thumb, exit by releasing thumb
- No trigger involved anymore 

To test:
1. Hold thumb button.  Teleport UI should appear instantly.
2. Far grab something and hold thumb to try to teleport.  You should not be able to.
3. Near grab something and try to teleport with same hand.  Should fail.  Keep holding and use opposite hand to teleport.  Should succeed.
4. Use the left stick / pad to move.  Try to enter teleport mode.  Should fail.e

This PR makes the following updates to hand controller lasers:
- Distance hold now originates from same place as search
- 'Search' color is back to blue

To test:
1. Squeeze the trigger halfway -- you should see a blue beam.
2. Squeeze it all the way -- beam should be red
3. If thing is grabbable, beam will turn purple
4. Purple beam aligns with red + blue beams

Additionally this PR removes the mapping of Oculus Touch controllers to SecondaryThumb, as this was causing issues in demos.
